src/descriptors/structs.json

```diff
@@ -1,244 +1,247 @@
-        "data_type": "Int64",
-        "comment": "Number of scenarios"
-      },
-      {
-        "name": "time_series_uuid",
-        "data_type": "UUIDs.UUID",
-        "comment": "reference to time series data; timestamp - scalingfactor"
-      },
-      {
-        "name": "horizon",
-        "data_type": "Int",
-        "comment": "length of this forecast"
-      },
-      {
-        "name": "internal",
-        "data_type": "InfrastructureSystemsInternal",
-        "internal_default": "InfrastructureSystemsInternal()"
-      }
-    ],
-    "supertype": "ForecastInternal"
-  },
-  {
-    "struct_name": "ScenarioBased",
-    "docstring": "A Discrete Scenario Based forecast for a particular data field in a Component.",
-    "fields": [
-      {
-        "name": "label",
-        "data_type": "String",
-        "comment": "label of component parameter forecasted"
-      },
-      {
-        "name": "scenario_count",
-        "data_type": "Int64",
-        "comment": "Number of scenarios"
-      },
-      {
-        "name": "data",
-        "data_type": "TimeSeries.TimeArray",
-        "comment": "timestamp - scalingfactor"
-      }
-    ],
-    "supertype": "Forecast"
-  },
-  {
-    "struct_name": "PiecewiseFunction",
-    "docstring": "A forecast for  piecewise function data field in a Component.",
-    "fields": [
-      {
-        "name": "label",
-        "data_type": "String",
-        "comment": "label of component parameter forecasted"
-      },
-      {
-        "name": "break_points",
-        "data_type": "Int",
-        "comment": "Number of break points"
-      },
-      {
-        "name": "data",
-        "data_type": "TimeSeries.TimeArray",
-        "comment": "timestamp - scalingfactor"
-      }
-    ],
-    "supertype": "Forecast"
-  },
-  {
-    "struct_name": "PiecewiseFunctionInternal",
-    "docstring": "A forecast for piecewise function data field in a Component.",
-    "fields": [
-      {
-        "name": "label",
-        "data_type": "String",
-        "comment": "label of component parameter forecasted"
-      },
-      {
-        "name": "resolution",
-        "data_type": "Dates.Period"
-      },
-      {
-        "name": "initial_time",
-        "data_type": "Dates.DateTime",
-        "comment": "forecast availability time"
-      },
-      {
-        "name": "break_points",
-        "data_type": "Int",
-        "comment": "Number of break points"
-      },
-      {
-        "name": "time_series_uuid",
-        "data_type": "UUIDs.UUID",
-        "comment": "reference to time series data; timestamp - scalingfactor"
-      },
-      {
-        "name": "horizon",
-        "data_type": "Int",
-        "comment": "length of this forecast"
-      },
-      {
-        "name": "internal",
-        "data_type": "InfrastructureSystemsInternal",
-        "internal_default": "InfrastructureSystemsInternal()"
-      }
-    ],
-    "supertype": "ForecastInternal"
-  }
-]
+{
+  "auto_generated_structs": [
```

Comment:
Are the changes here the addition of a second level? The diff is the whole file.

Reply:
The change is actually very small.  I moved the struct descriptors into a new JSON object called `auto_generated_structs` in order to match the format we use in PSY.  The contents did not change.